main.c

```diff
@@ -122,17 +122,49 @@
 uint8_t value_out = 0;
 #endif
 
-#if MICROPY_ENABLE_PYSTACK
-static size_t PLACE_IN_DTCM_BSS(_pystack[CIRCUITPY_PYSTACK_SIZE / sizeof(size_t)]);
+#if MICROPY_ENABLE_PYSTACK && CIRCUITPY_OS_GETENV
+#include "shared-module/os/__init__.h"
 #endif
 
+typedef struct {
+    #if MICROPY_ENABLE_PYSTACK
+    supervisor_allocation *pystack;
+    #endif
+    supervisor_allocation *heap;
+} vm_memory_t;
+
 static void reset_devices(void) {
     #if CIRCUITPY_BLEIO_HCI
     bleio_reset();
     #endif
 }
 
-STATIC void start_mp(supervisor_allocation *heap) {
+STATIC vm_memory_t allocate_vm_memory(void) {
```

Comment:
I think you should switch this to `allocate_pystack` and then return the allocation instead of a new type. The heap allocations can then stay where they are.

Reply:
The issue with that approach is that it proliferates `#if MICROPY_ENABLE_PYSTACK` checks throughout the code. That's how we landed on the `vm_memory_t` abstraction.

Reply:
Did as requested. Personally I liked the struct more.